File: 02-backend/spring-boot-ecommerce/src/main/java/com/paul/ecommerce/dto/Purchase.java
package com.paul.ecommerce.dto;


import com.paul.ecommerce.Entity.checkout.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Long userId;
    private ShippingAddress shippingAddress;
    private BillingAddress billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: 03-frontend/angular-ecommerce/src/app/common/data-send-to-backend/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    constructor(public userId: number, public order: Order, public orderItems: OrderItem[],
                public shippingAddress: Address, public billingAddress: Address
                ){}
}
